astyanax-queue/src/main/java/com/netflix/astyanax/recipes/queue/MessageConsumerImpl.java

```diff
@@ -46,15 +46,18 @@ public MessageConsumerImpl(ShardedDistributedMessageQueue q) {
         // TODO: Read full itemsToPop instead of just stopping when we get the first successful set
         List<MessageContext> messages = null;
         while (true) {
+            boolean success = false;
             MessageQueueShard partition = queue.shardReaderPolicy.nextShard();
             if (partition != null) {
                 try {
                     messages = readAndReturnShard(partition, itemsToPop);
+                    success = true;
                     if (messages != null && !messages.isEmpty()) {
                         return messages;
                     }
                 } finally {
-                    queue.shardReaderPolicy.releaseShard(partition, messages == null ? 0 : messages.size());
```

Comment:
Might be cleaner to split this code into line 45 above for success and exception handler for failure.  Code reads better than using a 'success' variable.